File: ClimbLog.Api/ClimbLog.Application/Features/Routes/Commands/BulkAlterCompetition/BulkAlterCompetitionCommand.cs
using Amazon.Runtime.Internal;
using MediatR;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClimbLog.Application.Features.Routes.Commands.BulkAlterCompetition;

public class BulkAlterCompetitionCommand : IRequest<Unit>
{
    public required List<int> RouteIds { get; set; }
    public required int CompetitionId { get; set; }
}

File: climb-log-flutter/lib/climasys/widgets/upsert_route_dialog/models/bulk_alter_competition_command.dart
class BulkAlterCompetitionCommand {
  final List<int> routeIds;
  final int competitionId;

  BulkAlterCompetitionCommand({
    required this.routeIds,
    required this.competitionId
  });

  Map<String, dynamic> toJson() =>
      {
        'routeIds': routeIds,
        'competitionId': competitionId
      };
}